### ISSUE TITLE
Software based feedback

### ISSUE THREAD

**Issue by ThrottleLP** [nhielost/obs-midi-mg#27]
When you trigger a command on the MIDI device, the plug-in processes the pre-set actions. Could this also be reversed so that the software can trigger certain things on the MIDI device?

Example: I have a button to mute an audio source. When I toggle the mute status in the software, the MIDI controller still shows its last known setting. Here it would be useful to be able to tell the controller via MIDI command what the actual status in OBS is, so that it also displays the current status of the software correctly.

Example 2: My controller has motorised faders. When I adjust the volume in the software, it would make sense for the controller to automatically move the assigned fader to the corresponding position.

However, it is probably a big effort to implement something like that. Would something like that be possible?

**Comment by nhielost** [comment]
Thanks for the feedback.

Currently, you should be able to send messages back to the physical device using the MIDI action. However, I am unsure how effective this is as you may have to test MANY different messages until get what you want. And yes, it would be a pain to implement this natively.

**Comment by LMWCoding** [comment]
I think we are trying to achieve similar thing, but not sure how to do it.  I have just installed the latest 2.0.0.  Firstly, thanks for the work, the revised UI is easier to understand.

I am trying to have a MIDI keyboard toggle the mute on OBS audio source, the keyboard also accept MIDI command to light up that button.  Idea is when I click MUTE on keyboard, the MIDI key press is sent to OBS to toggle the source mute, and MIDI is sent back to MIDI keyboard to light up the button, so as to indicate the source is muted, or turn off when it is not muted.
1)  I set up 1 binding.  To take the key press.  This will trigger 2 internal actions, let's call is internal action 1, and internal action 2.
2) Then I setup internal action 1 - This toggle mute on audio source.  This works.  great.
3) Then I setup internal action 2 - This will send MIDI command to the keyboard.  I can set the Velocity Value manually and see that it can light up my keybaord (velocity 127), or turns off the light on the keyboard (velocity 0).

Question.  How can the velocity value be automatically reflecting the status of the source mute?
